extensions/vertx-web/deployment/src/main/java/io/quarkus/vertx/web/deployment/VertxWebProcessor.java

```diff
@@ -599,6 +611,17 @@ void implementInvoke(HandlerDescriptor descriptor, BeanInfo bean, MethodInfo met
         invoke.returnValue(null);
     }
 
+    private static void registerForReflection(Type contentType,
+            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchy) {
+        if (io.quarkus.arc.processor.DotNames.STRING.equals(contentType.name())
+                || DotNames.BUFFER.equals(contentType.name())
+                || DotNames.JSON_ARRAY.equals(contentType.name())
+                || DotNames.JSON_OBJECT.equals(contentType.name())) {
```

Comment:
Could these types be wrapped somehow in some other type? If so, it would be better to use a custom predicate (which delegates to the default one once these are checked).

Reply:
I'm not sure I understand the question...

Reply:
So if you have a return type that is:
```
class Wrapper {
   private JsonArray array;
}
```
What you do won't work. While using the ignore predicate will properly ignore the inner `JsonArray`.

Again not sure it can happen in your case.

Reply:
@mkouba see where `ReflectiveHierarchyIgnoreWarningBuildItem` is being used, or the ctor of `ReflectiveHierarchyBuildItem` that takes a preducate.

Reply:
Ok, I think I understand now! So I don't expect nested values like this and even if they are used like this it shouldn't be a problem to register those types for reflection. That said, a predicate would be definitely a cleaner solution ;-).

Reply:
Is it better now?

Reply:
:+1: